e2e/util.go

```diff
@@ -0,0 +1,93 @@
+package e2e
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"math/big"
+	"net"
+	"os"
+	"text/template"
+	"time"
+)
+
+type Values struct {
+	Priv_sk   string
+	Sign_cert string
+	Mtls_crt  string
+	Mtls_key  string
+	Mtls      bool
+}
+
+func GenerateCertAndKeys(key, cert *os.File) error {
+	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return err
+	}
+	privDer, err := x509.MarshalPKCS8PrivateKey(priv)
+	if err != nil {
+		return err
+	}
+	err = pem.Encode(key, &pem.Block{Type: "PRIVATE KEY", Bytes: privDer})
+	if err != nil {
+		return err
+	}
+
+	template := &x509.Certificate{
+		SerialNumber: new(big.Int),
+		NotAfter:     time.Now().Add(time.Hour),
+		IPAddresses:  []net.IP{net.ParseIP("127.0.0.1")},
+	}
+
+	certDer, err := x509.CreateCertificate(rand.Reader, template, template, priv.Public(), priv)
+	if err != nil {
+		return err
+	}
+	err = pem.Encode(cert, &pem.Block{Type: "CERTIFICATE", Bytes: certDer})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func GenerateConfigFileForTest(FileName string, values Values) {
```

Comment:
just `GenerateConfigFile`

`ForTest` describes **how** this func could be used, and it's entirely up to caller, **not** implementer